src/Backend/Snowflake/Importer.php

```diff
@@ -38,18 +48,16 @@ public function __construct(
         $this->sqlBuilder = new SqlCommandBuilder();
     }
 
+    /**
+     * @param Storage\Snowflake\Table $destination
+     */
     public function importTable(
         Storage\SourceInterface $source,
         Storage\DestinationInterface $destination,
         ImportOptions $options
     ): Result {
-        if (!$destination instanceof Storage\Snowflake\Table) {
-            throw new \Exception(sprintf(
-                'Destination "%s" is invalid only "%s" is supported.',
-                get_class($destination),
-                Storage\Snowflake\Table::class
-            ));
-        }
+        $this->setUpAdapter($source, $destination);
```

Comment:
Možná by `$currentImportAdapter` nemusela být property a stačilo by si ten adapter předat? používá se jenom na jendnom místě.
`$adapter =$this->setUpAdapter($source, $destination);` a pak ho předat importu do staging table.

Reply:
Upravil sem